tests/linearizability/watch.go

```diff
@@ -99,10 +99,13 @@ func validateMemberWatchResponses(t *testing.T, responses []watchResponse, expec
 		if resp.Header.Revision == lastHeadRevision && len(resp.Events) != 0 {
 			t.Errorf("Got two non-empty responses about same revision")
 		}
+		if len(resp.Events) > 0 && resp.Events[0].Kv.ModRevision <= lastEventModRevision {
```

Comment:
I thought that this fields: 

https://github.com/etcd-io/etcd/blob/6c439aa9e2d4a53c63e8a28a227fc6e2f3a36ac3/api/etcdserverpb/rpc.proto#L835

determines whether we can expect continuation of the same modRevision in next answer or not.

Reply:
Forgot about this, however it doesn't seem it's exposed to client https://github.com/etcd-io/etcd/blob/6c439aa9e2d4a53c63e8a28a227fc6e2f3a36ac3/client/v3/watch.go#L96-L115

Reply:
Fragment handling is done on client library side. Client will aggregate fragmented responses until `fragment==false`. https://github.com/etcd-io/etcd/blob/6c439aa9e2d4a53c63e8a28a227fc6e2f3a36ac3/client/v3/watch.go#L644 

This means proposed change is correct.